tests/integration/sharding_tests/test_sharding_backups.py

```diff
@@ -200,3 +216,89 @@ async def test_rotate_backup_password(ops_test: OpsTest) -> None:
+    )
+    prev_backups = await backup_helpers.count_logical_backups(leader_unit)
+    await ops_test.model.wait_for_idle(
+        apps=[CONFIG_SERVER_APP_NAME], status="active", idle_period=20
+    ),
+    action = await leader_unit.run_action(action_name="create-backup")
+    first_backup = await action.wait()
+    assert first_backup.status == "completed", "First backup not started."
+
+    # verify that backup was made on the bucket
+    try:
+        for attempt in Retrying(stop=stop_after_delay(4), wait=wait_fixed(5)):
+            with attempt:
+                backups = await backup_helpers.count_logical_backups(leader_unit)
+                assert backups == prev_backups + 1, "Backup not created."
+    except RetryError:
+        assert backups == prev_backups + 1, "Backup not created."
+
+    await ops_test.model.wait_for_idle(
+        apps=[CONFIG_SERVER_APP_NAME], status="active", idle_period=20
+    ),
+
+    # add writes to be cleared after restoring the backup. Note these are written to the same
+    # collection that was backed up.
+    await writes_helpers.insert_unwanted_data(ops_test)
+    new_total_writes = await writes_helpers.get_cluster_writes_count(
+        ops_test, shard_app_names=SHARD_APPS
+    )
+    assert (
+        new_total_writes["total_writes"] > cluster_writes["total_writes"]
+    ), "No writes to be cleared after restoring."
+
+    # find most recent backup id and restore
+    action = await leader_unit.run_action(action_name="list-backups")
+    list_result = await action.wait()
+    list_result = list_result.results["backups"]
+    most_recent_backup = list_result.split("\n")[-1]
+    backup_id = most_recent_backup.split()[0]
+    action = await leader_unit.run_action(action_name="restore", **{"backup-id": backup_id})
+    restore = await action.wait()
+    assert restore.results["restore-status"] == "restore started", "restore not successful"
+
+    await ops_test.model.wait_for_idle(
+        apps=[CONFIG_SERVER_APP_NAME], status="active", idle_period=20
+    ),
+
+    # verify all writes are present
+    try:
+        for attempt in Retrying(stop=stop_after_delay(4), wait=wait_fixed(20)):
+            with attempt:
+                restored_total_writes = await writes_helpers.get_cluster_writes_count(
+                    ops_test, shard_app_names=SHARD_APPS
+                )
+                assert (
+                    restored_total_writes["total_writes"] == cluster_writes["total_writes"]
+                ), "writes not correctly restored to whole cluster"
+                assert (
+                    restored_total_writes[SHARD_ONE_APP_NAME] == cluster_writes[SHARD_ONE_APP_NAME]
+                ), f"writes not correctly restored to {SHARD_ONE_APP_NAME}"
+                assert (
+                    restored_total_writes[SHARD_TWO_APP_NAME] == cluster_writes[SHARD_TWO_APP_NAME]
+                ), f"writes not correctly restored to {SHARD_TWO_APP_NAME}"
+    except RetryError:
```

Comment:
Same as above, is this needed?

Reply:
Yes - the reason is I don't want to just assert failure when the `RetryError` occurs, I want to know exactly what the error was and what the count was, what was the expected count, and a helpful message.